src/memefactory/server/syncer.cljs

```diff
@@ -124,26 +129,35 @@
                        (doseq [t (into #{} search-tags)]
                          (db/tag-meme! {:reg-entry/address registry-entry :tag/name t})))))))))
 
-(defn param-change-constructed-event [_ {:keys [:args] :as event}]
-  (let [{:keys [:registry-entry :creator :version :deposit :challenge-period-end :db :key :value :timestamp]} args]
-    (promise-> (js/Promise.resolve (db/insert-registry-entry! {:reg-entry/address registry-entry
-                                                               :reg-entry/creator creator
-                                                               :reg-entry/version version
-                                                               :reg-entry/deposit (bn/number deposit)
-                                                               :reg-entry/challenge-period-end (bn/number challenge-period-end)
-                                                               :reg-entry/created-on timestamp}))
-               #(db/insert-or-replace-param-change! {:reg-entry/address registry-entry
-                                                     :param-change/db db
-                                                     :param-change/key key
-                                                     :param-change/value (bn/number value)
-                                                     :param-change/initial-value (:initial-param/value (db/get-initial-param key db))}))))
+(defn param-change-constructed-event [_ {:keys [:args]}]
+  (try-catch
+   (let [{:keys [:registry-entry :creator :version :deposit :challenge-period-end :db :key :value :timestamp :meta-hash]} args
+         hash (web3/to-ascii meta-hash)]
+      (add-registry-entry {:reg-entry/address registry-entry
+                           :reg-entry/creator creator
+                           :reg-entry/version version
+                           :reg-entry/created-on timestamp
+                           :reg-entry/deposit (bn/number deposit)
+                           :reg-entry/challenge-period-end (bn/number challenge-period-end)}
+                          timestamp)
+      (promise-> (look (server-utils/get-ipfs-meta @ipfs/ipfs hash))
+                 (fn [param-meta]
+                   (let [{:keys [reason]} param-meta]
+                     (db/insert-or-replace-param-change!
+                      {:reg-entry/address registry-entry
+                       :param-change/db db
+                       :param-change/key key
+                       :param-change/value (bn/number value)
+                       ;; TODO: fix this, it is not correct, we should query params here for the previous value
```

Comment:
you got me worried with this comment :D

Reply:
well I missed this one, good catch, I think this is actually a bug, since original value should always be previous one, not initial one, will fix it